### PR TITLE
Login page

### PR DESCRIPTION
**Pull request by rakeshrkamath**

### CODE REVIEW

File: src/components/button/index.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import React from "react";
+import styled from "styled-components";
+import { colors } from "../../variables";
+const Buttons = ({
+  name,
+  className,
+  onClick,
+}: {
+  name: any;
```
Comment (naikashok) [inline src/components/button/index.tsx]: Don't use any, give valid datatype.

File: src/components/OutsideClickAlert/index.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import React, { useRef, useEffect } from "react";
+
+const useOutsideAlerter = (ref: any, handlePress: () => void) => {
+  useEffect(() => {
+    const handleClickOutside = (event: { target: any }) => {
```
Comment (naikashok) [inline src/components/OutsideClickAlert/index.tsx:5]: Don't user any, give specific.

File: src/components/button/index.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+}) => {
+  console.log(className);
+  return (
+    <Button className={className} onClick={onClick}>
+      {name}
+    </Button>
+  );
+};
+export default Buttons;
+const Button = styled.button`
```
Comment (naikashok) [inline src/components/button/index.tsx]: Move all the color codes, font-family to variable file. Hard coded string to .json file.

File: src/components/country-code/index.tsx [hunk 4]
```diff
@@ -0,0 +1,170 @@
+import styled from "styled-components";
+import React, { useState } from "react";
+import icon_dropdown from "../../assets/arrow_drop@.png";
+import array from "./info";
+import OutsideAlerter from "../OutsideClickAlert";
+const TelephonePicker = ({ isOptional }: { isOptional: any }) => {
+  const [dropDownStatus, setDropDownStatus] = useState(false);
+  const [selectedCountry, setSelectedCountry] = useState("india");
+  const [selectedCode, setSelectedCode] = useState("91");
+  const handleSelection = (e: any) => {
```
Comment (naikashok) [inline src/components/country-code/index.tsx]: Don't use any, give valid one.

File: src/components/country-code/index.tsx [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+    setDropDownStatus(false);
+  };
+
+  return (
+    <FormContainer>
+      {isOptional ? <Label>Mobile No</Label> : <Label>Mobile No.</Label>}
+      <MobileNumberContainer>
+        <OptionSelected>
+          <OptionWrapper>
+            <Flag src={`/assets/images/${selectedCountry}.png`}></Flag>
```
Comment (naikashok) [inline src/components/country-code/index.tsx]: Don't use relative path here, give reference from separate file.

File: src/components/country-code/index.tsx [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+          <OutsideAlerter handlePress={() => setDropDownStatus(false)}>
+            <DropDownContainer>
+              <DropDownWindow>
+                {array.map((element) => (
+                  <OptionWrapper
+                    key={element.country}
+                    id={element.country}
+                    onClick={(e) => handleSelection(e)}
+                  >
+                    <Flag src={`/assets/images/${element.country}.png`}></Flag>
```
Comment (naikashok) [inline src/components/country-code/index.tsx]: Don't use relative path here, give reference from separate file.

File: src/components/country-code/index.tsx [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+  letter-spacing: -0.27px;
+  line-height: 22px;
+  outline: none;
+  pointer-events: all;
+  padding-bottom: 5%;
+  :focus {
+    border-bottom: 1px solid black;
+  }
+`;
+const Label = styled.label`
```
Comment (naikashok) [inline src/components/country-code/index.tsx:87]: Make separate component for label.

File: src/components/country-code/index.tsx [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+  width: 100%;
+  justify-content: space-between;
+`;
+const DropDown = styled.img`
+  height: 7px;
+  width: 11px;
+  margin-top: 8px;
+  margin-left: 10px;
+`;
+const MobileNumber = styled.input`
```
Comment (naikashok) [inline src/components/country-code/index.tsx]: Make separate component for input.

File: src/components/country-code/index.tsx [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+const OptionSelected = styled.div`
+  border-bottom: 1px solid #a4a4a4;
+  width: 90px;
+  display: flex;
+  flex-direction: row;
+`;
+const DropDownContainer = styled.div`
+  z-index: 99;
+  position: absolute;
+  background-color: rgb(210, 210, 210);
```
Comment (naikashok) [inline src/components/country-code/index.tsx]: Move color codes to variable file.

File: src/components/textbox/index.tsx [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+	const [viewState, changeViewState] = useState(false);
+	const [focusState, changeFocusState] = useState(false);
+	const handleViewIconClick = () => {
+		changeViewState(!viewState);
+	};
+	const handleFocusChange = (e: any) => {
+		if (e.target.value === '') changeFocusState(false);
+	};
+	return (
+		<FormContainer>
```
Comment (naikashok) [inline src/components/textbox/index.tsx]: Still alignment issues in this file.

File: src/components/button/index.tsx (hunk 1 above)
Comment (rakeshrkamath) [reply src/components/button/index.tsx]: I haved Changed that

File: src/components/button/index.tsx (hunk 3 above)
Comment (rakeshrkamath) [reply src/components/button/index.tsx]: Ya have Moved everything to variable file

File: src/components/country-code/index.tsx (hunk 4 above)
Comment (rakeshrkamath) [reply src/components/country-code/index.tsx]: I have specified the type

File: src/components/country-code/index.tsx (hunk 5 above)
Comment (rakeshrkamath) [reply src/components/country-code/index.tsx]: Have used variable for the path

File: src/components/country-code/index.tsx (hunk 6 above)
Comment (rakeshrkamath) [reply src/components/country-code/index.tsx]: Used Variable

File: src/components/country-code/index.tsx (hunk 7 above)
Comment (rakeshrkamath) [reply src/components/country-code/index.tsx:87]: Created separate component for label

File: src/components/country-code/index.tsx (hunk 8 above)
Comment (rakeshrkamath) [reply src/components/country-code/index.tsx]: created separate component for input

File: src/components/country-code/index.tsx (hunk 9 above)
Comment (rakeshrkamath) [reply src/components/country-code/index.tsx]: moved color codes to variable

File: src/components/textbox/index.tsx (hunk 10 above)
Comment (rakeshrkamath) [reply src/components/textbox/index.tsx]: Ya solved that

File: src/components/OutsideClickAlert/index.tsx (hunk 2 above)
Comment (rakeshrkamath) [reply src/components/OutsideClickAlert/index.tsx:5]: Ya specified

**Review by naikashok — APPROVED** [review]

**Pull request merged**